platform/ios/src/MGLMapView.mm

```diff
@@ -3762,61 +3768,69 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
     queryRect = CGRectInset(queryRect, -MGLAnnotationImagePaddingForHitTest,
                             -MGLAnnotationImagePaddingForHitTest);
     std::vector<MGLAnnotationTag> nearbyAnnotations = [self annotationTagsInRect:queryRect];
+    BOOL queryingShapeAnnotations = NO;
+    
+    if (!nearbyAnnotations.size()) {
```

Comment:
If there is both a point annotation and a shape annotation beneath the passed-in point, this method should return the point annotation the first time this method is called and the shape annotation the second time, as long as `persist` is set to `YES`. That way the shape annotation remains accessible even when there is an overlapping point annotation.

/ref https://github.com/mapbox/mapbox-gl-native/pull/9984#discussion_r144387593

Reply:
I’m OK with this detail being tail work, by the way. Overall, this PR is looking good, especially after the last few changes in core.